### PR TITLE
support composite mark's field name with space and punctuation

### LINKED ISSUE #4733: boxplot doesn't work with long field names
Field names with space and other punctuation will return errors

* `"Life expectancy"`
* `"Population, total"`
* `"Fertility rate, total (births per woman)"`

See demo [Vega-Editor](https://vega.github.io/editor/#/url/vega-lite/N4KABGBEDGD2B2AzAlgc0gLjMSA3ZApgO6baRHIAmALgBakAsADEwDRS0Fq3WkDMLAL6DW4KJQCG1CaRzwJAWwKlIk6QFoAbJugBGPgE4ARgcR8AHAHYmiaACZNiI5cQ3NlgkYKVLuo5BExSAUJACcAaxUjWAAPAAcAG1heUQhIAng4SmR4dCxQCDS4JNDZSGoATzjlLEh4WAUciQTIdkgUAgTKFQAlAlRkBADUwsgYssrqlXrG+Ra2jq7e-sH4YbE0itkN0cmaqABHAFcJeGpkaXPcZRHCqEXu2oAxAlDzhIuKsFCpAnZqZLNMAACiMyDetAAzmBqqEwEQGqcAJSQHaCMSBCBBNQSSEEaiQ7Z3VRSCRaHT6YymCzWWwOJwuNweLw+PykADaOwgBTuowAMshEAQwAR4gRoNJMlssJoAKwAOiYdl0BlVavVBluvMgAAVYHEjgkpKt-oCWlg7ExNCqmJZZVriS83sgPpVvr9TdIEiCwRDobD4Yj4CisHx5bomJquaM+gMhrUAIJKULIaC41G8sCY3k8zOQAVCkViiWnaDSsDWeXmXTWSx8Gu2+uWB2jPUGo3nBCe5qkCN2SyaBiWZvRtJO96fd3UP5gAFen3guj+16BkLB-jyhiygQsXd7lh8FtpWOrFQAeWgBFOyBk0ezd1z2oLwtF1RLUtIsoY8rsDAj+73XQjygNtDWNLtZzNUhLDsS07S-YDIHHF1Jx+adu29UFFyhGEVwRNcQzATRFRHTNjxWeMoCTV5U3TO8W0fYlnyLN9JTLRgwxg+1RxA-UwM7eAMNIbRaz3RDkNdL40JnOcgSwv1cLhfDkU-eVLCHDVNKjMioBPSjIATSEbwzXl70KRj+UFF9izY8tNAMKtNHMOwDAcZzXPcbSyN1PiOxNSCvUYJzdAYOw+FI7yJNQj0Ark30l0U1cVItKtuJ0yA9LWWoAFEjlCfVlHo6MLLSZjX3FWzhIc+sWF-QwlQYQxENAvyINk80KzsasGAMFhxNeCc3WkjCFwUgNlPXFLwsQzKVFy-KpiKzMSqgMqbNLcs-3DdTmt88DBNijqGB6hhzCtJhzH651JKnGSzVGhLxqDQiFV6zQAP3ICeIyiisqoozb0zMzuW+tbWI2xhzHlOUvvSlr9qErArEsVzZV0NLIoGlChpi9qHpwp6CNU7RNBm36VEM4ylpzUGrJYiqIZlBzdDC3q6rZ47dvbBHDs-JgVQMaw2G+qKcfQw78eXJTntIOx5Wm77ZpyvKCpMu5gewWnC3K992JlSxoe0SM7XewXZVJ774YExGwFRlhZRgq7Bqk3H7vkx68JllKWFh7ylf+qmgYYrXrPBj8sG3RV+bO07LBj8xLstvbrd5i1ZX5sKdpFrGbuGiX3YJz2iZlRUDAxvN-YMgG1cKDWVvzOmdcqiODYuuwuf4-z2v4at1N0Fmnexl3xbxgupaSyaiLUpyybjP6DOTWjCWph8Q-p3XyzreWPPctyXMPJPuZT7usH707tHrQfc9d+cx8SibCLDAfFfJ5WFsKoPirXpvGaI4imFlLKFye8d4d1agdE+YB+4GHMILXcV9oojzdvFQu0ti5gDlkA2ep436qxXuZb+61w5gC3NDCwH09zl21FbLuUEsDMHMHKJq2drqILurfFB48H7QR-LoC26VK6U0BqZYO6UwYM2IfZcMkYQEgLATzSBAgtzpz6iw52t0Rp30JslDBjlsH6Xmngz+y1CFhz1hWXQ8t9B-gYJoPgNi7HyOPnQsA25+zOQinmUWw92FxWwlwr2ujOYvznnNFWi1jE0zEY3Ih5jSF8Hsk42hgULTqRgnwS+aih4aPzpw++gTiIJPehQ3cVDiSCOrvgkG0TtaxM3pYlmTlgFNNcr4JJbUXFAOYGJLJ18kEcP8fk9BctEkhJwVAQxESRFfxqaHCR5ipHDkbA2OsNZ+7tIgS43QZ13D-mFulbxOTR55O0ZPMMcoGD6PntRFMaZl6RNXrM9ezcwB8DlsAjZNtYEGDsWjBBYtfGYROUXHRBstxXIppUh5BCnk-0ke8uOWl1T8O8jQjpKSoHvQTqog5Oc2GaOBWgnRxFTpeQrq-CZ4SP7TJMbCupQVwzkJKQAz5qcoH91-AnROuLWEAoJYM05hEDZdIhYmRAtzhHq1Ed5cRG9SDwENAkVlkDLT8wPjy9RedjkCpBZPBVCQlVjP0heK88BA40qiTKmJZjyz6sNXDZOySeyn3ses3p+Lck6qJXqxVorKXvxrhAOupj5nlgyT+P5h9O7ouda4wc9ZnL-J8fysauqhWbjjn6gy4raKBqzNKvMsqXmaDlv3MprZHUxo6oAjJZLtSHK1cgr1E9CLflCuW8ioTEyLzuXm4NdKbXCVem0qN4CbbbL7paJNRym2pu9a2tSjDmX2yzUIvtBanzWtDUOw2AskV1uJGizZGKvzKl9l4vFfLPVzpbYwKs7cjXXJ7XRaF1SrW1MHTKKGspzDm2Nn+wWKK8xHpttYIc24D2jAbTfPxN7uEpVGQIilkBJnUqlTM99cy5UyksUwPgHbeJHydR1CwZdDBAfrZe5N16PbztUsjfdapV1QotY8zDzzf4lqrNWZVWzIxAO5ZjXl1HtVwcCW2suq7n33NYzC9jcLzFo0coJ4Dlbj2xr4b+fs07G0DLE+g1uVhl3nu1BU816HaXyfpfQhU5hMkOqI1WyGAs0k6Zg0C5t8Gp5cWY+Z2uG6mJbuw0RXD-NGOSdHQolxYVIzMI1dk3TsHaO3vofLDJUmaK9qqZrAd279ZqQuv+02JteMYurDA+B7qr2ieS15kZnjTPIdQ+ujDhagvFoctYFUaNlRlxZsqUrsaDDo3sbAtz-SkuoJS7ohCj6wkBuy-XItv9ZQORgSZw9ambauStJcqrInZ21YKfKAjP0u1UWky1yzbWP15deQbfsG2K2OfUx1OqmmH3xb6YCyWQziU-kTXNsVEqruWpu1hl5drBtHWOjioTmr3O-cFfK31QOLuZZfbJt94OOPEPsSduwq3GMUc2y9m2sE3kJ3Gz9rRabhLQwa+U5Da7FshuC6FRlZ1mWnZA2yhJBh7E9Wpymo76DiKM5jMzy7rPcvBfevLbc0P+CwXNjPfbM69Oi50WGM6T3O3jJQ1S0HbGccKfLOnXdSuI6-kAZ9+HCXEe07oyXa0p3BE5rTMbuTpvrNgCh5F5xZW8Pqvt99kXU2vP+6Q+dheGOZMWbB5u27cu5a2Il2kXnkDe7gvV4ljz+n-t2ZJ5LmPLPX05as5+oiDlfwmyK5GFT1Cttsrju4cKe2vsepqxHwJct7N+ya0bmXle7v48anupFPPm+QNOsAqnufHeEum8RFyevdLM495K-zrWk8Q9-ltLqeyKHF4z9P6LreIwn6gNBib+eteT2ImXFy4XV2b699j3fuO4lQ1clzkpHfUUz8MVLQwoIxhcaMe8xduM3cN8Qdh8fcq8D9QordXEYJM59lQ8u9DtID-sfkhZudfMt8g0AtLJk8XktovxikKE18fIyc2U0YBw3JwDu8Al0EFRBx0919S8394DP8zdoIGkMkVkmxGwUCaxepepIxmDsDWDtcqx+5l0mAADyVS9pdy8lt2tf4o9AC6DIFW9Tsb8acl9I9Udo8Dcbkl538K8EDR8SVYFl1lCm9dCtl058MLppDNccDJ5bNHCmduC4D1C2cXkLBFQVQUCdx0YBcPDJtZCH8I1X8AisdrC+DfcHYAcp9nCMVhw+ES0DCqMNcYi-tvDpE7cVDzC1CkiNCyCVtxc0ZwsMjo1XtexBwzoLAQ8L1hMCi78vDCJiIzpOCztyi48rCqi99iEtpjp7JicxDhsWYaDDDw9Yjei1IdtmUaD3dEiE8TcUjEDcMYExCulnJI1O9qsZCijlifYEjc1eDAtqjxioZ3pSinDGjx08M0CMCOiEdb8kc6cI5uMrjPcbjSCxiFkmATs-1mVfDnsXi2V9AMklCPjKNOi88fjncp4twj8KEYDVDhigTSpNC8c+i5RwjAFYF58TiDtPClieF7J1jYDrjAjZdi1vwsU5Rf0Ekf1ZsHMYSZ8QoekKSujUTpt3lsVl12jGsY9ms8TVoCSFkdda1piA9iMBC-x9A8jkTF9PNxNwxyt6iMsJV49t9rsdi7ttDVNMiNNOUHBojujqSsAzSJShiDSRigjOMHJrRG9SceTOkYIBdETHR8iUSndpswxsTzCWMtjvcTTgtKxRs44lCrAY5INT8LSOofklQWAr8kJAzNSC9J45YpC0dDcFtGSR8YzS0BxOSOT2SBjM8XEtwUZ3pxSAyNTvjgy6sTsBjK4pTSybD2dvw8M-8KEoSUzvSysEkEkbShSvMSVrQASiD80d9biQTbVTCdCxzY1ik3UBSgzjDAkHS-CnTLDpSG47jFNLE45aygDY03kDBmBkCF82y9yDNwSwz9ILCstezoyXlI4woqDhyUCtxlF04pz2zjsYI6ScTnSTzlt7jFQIT-8UCHBVQmFmyoMcynytSoDfAsyNiGTKjXTxi1snJwsRzCMNy3tGDvlQLnz-s29wsIszD3yKjIyP9lyv8w1xcyLaCKLZYa0woaKsKdF2DFciyEweCvz2L+DQwoY-xPToSx1YTDB1JhtBK8zljqKxKJKCKmTOMDY2YOZ6pfxuK6yMU5iRsBiFiIC7SoF5R9iiyeydKyzyDvwuoGjFLFEeoaxuKrKWDzj6dfB8CSk3zrltLWLkipLfcQi-xJDEKlSnMLRlQetM1HyjChK4jySB9-D8LwrRiOKUcDVwjbE5zUrFj-L7S1yyiDEh9JLgT8qI5bMszTKbyBcLAszfKzjkcS5bFIKDcTVrwFz+1nLOMoZys68zYSruSPL6zQofl7E1L79CI5ZHYHKaqnK+yfyV9Acpqossj+j2qMK0r1LiYHYGKQrzxLwBqXTdK8cHIf0pj90mrryjpVRtkfLDqyquqSF5Y7Mzr5zrrhq8cCz8NwjPIuoFqejVIHZn9GNzrgccqjTE9Iqq87FoZjj1zpqMUEktxikIabK5Y7yJ8kUuzmcIzEbtjka7tSFzBGpSKUDtAgEupLKPrrLyrvMhctLNjyaozKbgsDyFLdrY0wpDB3rWyjrFqCr7UsqDdHLcrCKFkoY+EjLaaGoTLnrSA7MmxCc8a2b3k0L9dqqSz1rvytDKrnjMabzCc4dPiHdMLjqKrCqxKWLua2K6rpKiIwSrQhyPp3LBaOoyTENMDTiqS2auKYb919Tjzar8Szy7IwTBZ6b3AHydzcyJaS4AFVUCCiz+qzVBqSCY6VyGVrAnivSLajplRbQhwdavq+iflI7Pzjbebgjvwxt4qmjQwHZ1JCyU67a06p4z5-qYLZSw0DZf01bUzhI+FmBkzr8Wa-KvrXoBtOaEbiCly3bfd5drS26bYHZtkwjSrWaa75YYJjN67Mc5abq4lS19BwjGFXcZ7syxbPrfip43luLBFnbV7jSm798V86w6bt62VLQ2qFYe7xbIasAwVTqX8nbcTo6ZTY7hJH9rQWkQF9byKy7+BbQ-11SvjwGbKFQ1UGLCCAaNqRqo4RDVlax0GeLMGsBVRfr3Bq6X6CyS6S8Za1qL7AbzEt5QpKH+GxD4Is4wHn60TXKH7uzOGXaIr17EDvw7yHqkUnqJ6LQbR7I8NmG0TH9+xFC4aA48616C76qSEoYlQ2HRy6HOpYEBx5i57OqX6V8gFT6xKyav6kbZG7tFkfketut+skLnIl1-T0Kn7D6X7OJ5KDb55ZbpG8r3b+aLG-bIZjpzHZ6Qn56X74muCOGjauGyHiFMnaHEnVG+Sp0D70m0SddtASGh7EGZRvwaQxCS0jiUnH68HRHpsFQYqz7DS3GKaPHgsgE0aaDmq0y3lCdma0n7G0SDYnISbS9XHFzv7+mfyoYEk7R2S2TtwryVHbZfBQp3pNHptvxzYWmzMDGlmjH3bbF5ZToUDqwAFYdDmvN9LbFnGmLQqubemebln98a8c8drA9Y0-xhannwKrBiHs7Lrc7SGTbiElMXJtneLT4QpfwawDrJmQ6vrjmUZqn4HTzC6ZQwS3KkL7IPTQWXzjLcXG6fm4XiIWZ+j+GawkKawYsgmxw7HMWX7OmSKYH3nIU-MvnXbLnfdGofw8NeWMaimMEWYLo66ympmOm7LJrpb3yFmhq8mFlOtGFd5UHEXLHlRLRBYH6OrOW0TLE7ys1onBWZHhWUawwHBFGtJlGkWoEkyt6RHQm0SFR-6qXcnYW4liJHi7m0ZVtbGMXCiF6qxSM9Tl7AS8XYLzF8ddkxCZX6waGTWI2HHNxzo1jB743h7GARlZXYaxDjoaamX5XTXpsHsMzc3Y3zn3HbXTSzbS6pWp7GKg7KTM20SCnJGcmYn5a7IHIaaqHllawkLhxf0QLK3u3hSTtcLB9+3rXYmN7ZLbFlblbwjKRrByWdF3kJHF2jFqWm25cW7wbAHIE+Fhw5WPXymjm1IVq+XEw1X86EGCWPayEaGRnexTo76Jm2nPWQyqxk6VWompHl3B3+BiKibNJnXLH05VRwp02OXZ3nn5Y83j233jHwoqwWUL2XEEltlVtcHbb8HQ67L376S43MP8XjHVtNwqn8Oyt0ZfArRd24j-nQP+WG2+mT3giDZlK7mYErB2P00lRerVWBXFnG2sP3aupwxVQUCgE7E6xkPw3bTyO8DdGMO-Wf7JFv0n8d5mkImMGpW3k5Rp3b2FWvNzlTnD2pldOaXzECnv3T5iqWmM2NOvre3kMc6pP1X-XNowxf3g3qw1crOq2ZyH29Hs1PnpPePZPUioZ1IgFfHet0bzSXW5QBcMukSAO72our23muP4bqPHO+OVthUTPCnAWYctnRP6cBdKOY8-OePvmKvJFzWO3Mv9Wvxuk2XUn8vrPAkwxKdFCaG+2j3yvEuq8XP1bEqtwPOUOvOMmW32HDapuB3L7NpVnhtws4OpWy5AE8OIvUPjtDOJWqrrkX3DGZvPGdcWnXOTGAEcWZ2VuKnoYNHVql34v2u7v2dZKf0xCwo3K3kGvIHFRmvzCwqtvuGw1CHCd9uUC6wE57ISOw9AOouy4of3yYeIPtvC21IQokKG9rQBK3vpzjsAEJPrv-PX3aOrnH9hn5uPbCdglTv3vl9wwOan2qI8ffuhX-vIc1uEnavGBGo1OhvIv9yResnjUoW6fbuGeoqFQVRHXYOxDMyBBtzO3BSwK2C7LVtfXYeNXzcHiN3DLepx6XWUZKsOfKeoD3Es1WuYW9OeHLE-yqzNnQHJWxeLRwptlra8vSP2mvMByJvfOFe2vBflfEDW4v2Weax1IpjweKw0sF2oKo6aOE34fIffa-fXkLArBBxU+oGcePmV6BebWhff4Ek7K7RAK0Zwpqwlv1OHf-t8Ny+KZ+eAu3fVzHaAXlTT4Dx0esDpf0EfOWuo-XenPN5QyepxqTZmedn8MW+4tdfdz0qlrUpLXwOq+V3ED6tHuWfAFvKBvWmQ-MfAkHshxiurvuOZ+Ou4kFQHB8+h+SEeo24oiKf9e6Ljf8e4ektFAj8hA420MeBXGXgPxK5QAXeNTd9vLkYQwcNQb-BKuylvJL17ev-YoiJ1gbQV82tTEhJYlgSYkAIKA9uhgn5hWhhwqfP+jF1gH4D4BOuV7oP1QE2gmao-YOmdygIxcPy59E3oF0YAzMBcSPJjhpkzLox0YNA+vjTwpif19+kHFuI5D1ZSsHA-YAQNrR-60VJ4BsEKNVzOaP8a+kiGZvHHjKxweMogt7HNRS7n9PO7ffMnZT0H2c0M-AvvsJDlhnRYEsiFpGQK+S-psi6LKXlwLkJ8Jd+P3XvrP2EjJc1e9eQDEhVX528N+qdCBt9TlBzMjyDdabrH08Z7FDAjGa3pYyHAWBRagQznljxHS88q4ivC5oYIWTsFQoKtdmF9xYHkC3qVoXqBwK7alDe8NzUIZtwAGm9CeHyCwbLFaqhRqutgrAYREVpGY62FQ8SnF3CFP9NonWU6Muh8GwkLog4AIZfwgEvkYYzvafnAOMZzcdmI2BOpoK35ADvufQ+QQTwjhglsa9NdtJLx2HDdHeoUO-o6Vx4LD6eOfHuD+HJ7NCbYbyKgTz0SG91khj+G9tANi6V9FhNQ-vlLR65Ss-wQsVPpP2h4-Clefw0MA8PKG+93+sEQWAoSkEswgqH0GQaV2j7V8shcuESjAmM7AJlBBfKeoOCD4tkShdgl6NPH-63DABTMOyp7w2ZVkH6T3ZgE5H4yp8wwShXoQ5xcERCv09fEwcqIOIjtSmmArQdv3sSRhwsaQjbnKP6ECD6EzMYRgSNYE1g76UgovgcNNRVCZOtI4Io-moHDDUkgCO8v+1eHj9QU8RetgYIdErZ3kkgl0VAhS5ORAEUgpQpSL55Yjqh-ovHJYnsjMjCRR3dGK305GTCAqGAmEfMLhG-CC2EcaqIHWREF8YE6kU6GG3TGaioaf4XkfCLjHOdZeNXd-mjG6zojGxH9OBtn3zGoF5YXUS7ubSlZklzhGoy4SXC05rDumfonEX7kbFij+wxrZblyKuFzC5BdY6cT+kVAJ8dmZ0O8guLb4ZiIesELvt2k7GZDpxp0BTrkMerI8+uXlKQbSR07yilhn4d0gdwL59ifk2w8AW8LoruFcBWfM8d2K3j2xisAGAcMj1Oi2JzYqfCsrKOcGGjXBEcc1qx0hKN86wPWUvgx3JE+1axeYggebHDBYJgxloIzBYCkFfhsJn0XCdiO7GisnI-5ACMvxdb-hrBHo78V6IypsdfRRw92l+AKxbiXWag4kQ3wuH21+6VdFxnaIS71iw0nEFyIBTVCWdwRZHI+uFHDpIpqJsY88US2HFmiWhP6ani8PYlBC4iqXR8QhIVEhZs2TEyxig35iYSEk6krSHQMOEMDjGXGXInc0gnMDlJofQJC-zsQQsVxp4p8QiMJ7idAKaLB1lILvJ2dsqZXUKTJM-BtoUJcVIEbCSbKE5xhi4g8VPCN6STqRB-ZtlAOLHNiYIX4sfiZMIgYjmKIUiyc+IdpIiBxBfa0DKNEl90ap88egV2IIGitXqswvSTbHcAJxlQxQz0VVMa6KduJbk92k5B-Cqhtw3vbLoBQdajS9xlY0cbol-BwSpx3YwZq-2AEOB5q7UyEYKO+SMZNJ9o6ce4EXTH8dmhONqrlw5HjSuhwyf4v+IyGJT1xo1EGsGLHq7ijJlU16f9kJw8CbuWk2iSShrHBi7EKfE6TZR0G6T7+tQbqYBPwk15RK6UrPCFCcgSSRxYkwNnwm04fS+B9UsKQWMh4kD9w+Qszg4GsAlp1pL0pcSXFGyXTpJ10lfJykWle83x7-N5ColT7sEgxwUvAT1Pfb45COSnFlmqUwmxw-qBU3ab1NV6swS2wYpRJBIqmcDgZ2g8MMLOzE988J77JTObHWGKI0k-OTCSx3llzCe+OwAALoYgQAggIAA)

### PR DESCRIPTION
**Pull request by chanwutk**
fixes #4733

### CODE REVIEW

File: src/compositemark/boxplot.ts [hunk 1]
```diff
@@ -212,12 +212,12 @@ export function normalizeBoxPlot(
   let filteredLayersMixins: NormalizedUnitSpec | NormalizedLayerSpec;
 
   if (boxPlotType !== 'min-max') {
-    const lowerBoxExpr = 'datum.lower_box_' + continuousAxisChannelDef.field;
-    const upperBoxExpr = 'datum.upper_box_' + continuousAxisChannelDef.field;
+    const lowerBoxExpr = 'datum["lower_box_' + continuousAxisChannelDef.field + '"]';
+    const upperBoxExpr = 'datum["upper_box_' + continuousAxisChannelDef.field + '"]';
```
Comment (domoritz) [inline src/compositemark/boxplot.ts]: Let's use template strings here.

File: src/compositemark/errorbar.ts [hunk 2]
```diff
@@ -512,26 +512,26 @@ function errorBarAggregationAndCalculation<
     if (inputType === 'aggregated-upper-lower') {
       tooltipSummary = [];
       postAggregateCalculates = [
-        {calculate: `datum.${continuousAxisChannelDef2.field}`, as: `upper_` + continuousFieldName},
-        {calculate: `datum.${continuousFieldName}`, as: `lower_` + continuousFieldName}
+        {calculate: `datum["${continuousAxisChannelDef2.field}"]`, as: `upper_` + continuousFieldName},
```
Comment (domoritz) [inline src/compositemark/errorbar.ts]: Please combine the strings into one template string.

Comment (chanwutk) [reply src/compositemark/errorbar.ts]: For consistency with `as` clauses in other transforms in composite mark, should we use `'upper_' + continuousFieldName`?

Comment (domoritz) [reply src/compositemark/errorbar.ts]: I don't understand the question.

**Pull request merged**